L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -670,48 +798,66 @@ void HGCalTriggerGeometryV9Imp3::fillMaps() {
+      //Stage 1 links to Stage 1 FPGAs mapping
+      stage1link_to_stage1_.emplace(link_id, mapping_config.at("Stage1Links").at(link_id).at("Stage1"));
 
+      //Stage 1 links to Stage 2 mapping
+      stage1link_to_stage2_.emplace(link_id, mapping_config.at("Stage1Links").at(link_id).at("Stage2SameSector"));
+    }
   } catch (const json::exception& e) {
     edm::LogError("HGCalTriggerGeometryV9Imp3")
-        << "The mapping input json file does not have the expected structure for the Stage1 block";
+        << "The mapping input json file does not have the expected structure for the Stage1Links block";
   }
 
   try {
-    //Stage 2 to Stage 1 mapping
-    for (unsigned stage2_id = 0; stage2_id < mapping_config.at("Stage2").size(); stage2_id++) {
-      for (auto& stage1_id : mapping_config.at("Stage2").at(stage2_id).at("Stage1")) {
-        stage2_to_stage1_.emplace(stage2_id, stage1_id);
+    for (unsigned stage1_id = 0; stage1_id < mapping_config.at("Stage1").size(); stage1_id++) {
+      //Stage 1 to Stage 1 links mapping
+      for (auto& link_id : mapping_config.at("Stage1").at(stage1_id).at("Stage1Links")) {
+        stage1_to_stage1links_.emplace(stage1_id, link_id);
+      }
+
+      //Stage 1 to lpgbt mapping
+      for (auto& lpgbt_id : mapping_config.at("Stage1").at(stage1_id).at("lpgbts")) {
+        stage1_to_lpgbts_.emplace(stage1_id, lpgbt_id);
       }
     }
+
   } catch (const json::exception& e) {
     edm::LogError("HGCalTriggerGeometryV9Imp3")
-        << "The mapping input json file does not have the expected structure for the Stage2 block";
+        << "The mapping input json file does not have the expected structure for the Stage1 block";
   }
 
   try {
-    //lpgbt to Stage 1 mapping
     for (unsigned lpgbt_id = 0; lpgbt_id < mapping_config.at("lpgbt").size(); lpgbt_id++) {
-      lpgbt_to_stage1_.emplace(lpgbt_id, mapping_config.at("lpgbt").at(lpgbt_id).at("Stage1"));
-    }
+      //lpgbt to Stage 1 mapping
+      unsigned stage1_id = mapping_config.at("lpgbt").at(lpgbt_id).at("Stage1");
+      lpgbt_to_stage1_.emplace(lpgbt_id, stage1_id);
 
-    //lpgbt to module mapping
-    for (unsigned lpgbt_id = 0; lpgbt_id < mapping_config.at("lpgbt").size(); lpgbt_id++) {
+      //lpgbt to module mapping
       for (auto& modules : mapping_config.at("lpgbt").at(lpgbt_id).at("Modules")) {
-        lpgbt_to_modules_.emplace(lpgbt_id, packLayerWaferId(modules.at("layer"), modules.at("u"), modules.at("v")));
+        unsigned layer = modules.at("layer");
+        int subdetId = 0;
+        bool isSilicon = modules.at("isSilicon");
+        layerWithoutOffsetAndSubdetId(layer, subdetId, isSilicon);
+        unsigned packed_value = packLayerSubdetWaferId(layer, subdetId, modules.at("u"), modules.at("v"));
+        lpgbt_to_modules_.emplace(lpgbt_id, packed_value);
+
+        //fill subsiduary module to stage 1 mapping
+        module_to_stage1_.emplace(packed_value, stage1_id);
```

Comment:
There could be a silent bug if in the json file one module is split on two lpgbts and each lpgbt goes to different stage-1 fpgas by mistake (which shouldn't happen).
It should be checked that if `packed_value` is already in the map, the `stage1_id` should be the same.

Reply:
thanks - I have added a check